src/Api/Tools/Models/Response/MemberAccessReportModel.cs

```diff
@@ -0,0 +1,150 @@
+                    {
+                        CollectionId = tCollect.Item1.Id,
+                        CollectionName = tCollect.Item1.Name,
+                        GroupId = x.Id,
+                        GroupName = groupNameDictionary[x.Id],
+                        ReadOnly = x.ReadOnly,
+                        HidePasswords = x.HidePasswords,
+                        Manage = x.Manage,
+                        ItemCount = itemCounts,
+                    });
+                accessDetails.AddRange(groupDetails);
+            }
+
+            // All collections assigned to users and their permissions
+            if (tCollect.Item2.Users.Count() > 0)
+            {
+                var userCollectionDetails = tCollect.Item2.Users.Select(x =>
+                    new MemberAccessReportAccessDetails
+                    {
+                        CollectionId = tCollect.Item1.Id,
+                        CollectionName = tCollect.Item1.Name,
+                        UserGuid = x.Id,
+                        ReadOnly = x.ReadOnly,
+                        HidePasswords = x.HidePasswords,
+                        Manage = x.Manage,
+                        ItemCount = itemCounts,
+                    });
+                accessDetails.AddRange(userCollectionDetails);
+            }
+        }
+
+        // Loop through the org users and populate report and access data
+        var memberAccessReport = new List<MemberAccessReportResponseModel>();
+        foreach (var user in orgUsers)
+        {
+            var report = new MemberAccessReportResponseModel
+            {
+                UserName = user.Name,
+                Email = user.Email,
+                TwoFactorEnabled = organizationUsersTwoFactorEnabled.FirstOrDefault(u => u.user.Id == user.Id).twoFactorIsEnabled,
+                // Both the user's ResetPasswordKey must be set and the organization can UseResetPassword
+                AccountRecoveryEnabled = !string.IsNullOrEmpty(user.ResetPasswordKey) && orgAbility.UseResetPassword
+            };
+
+            var userAccessDetails = new List<MemberAccessReportAccessDetails>();
+            if (user.Groups.Any())
+            {
+                var userGroups = accessDetails.Where(x => user.Groups.Contains(x.GroupId.GetValueOrDefault()));
+                userAccessDetails.AddRange(userGroups);
+            }
+
+            if (user.Collections.Any())
+            {
+                var userCollections = accessDetails.Where(x => user.Collections.Any(y => x.UserGuid == user.Id));
+                userAccessDetails.AddRange(userCollections);
+            }
+            report.AccessDetails = userAccessDetails;
+
+            // Distinct items only            
+            var distinctItems = report.AccessDetails.Select(x => x.CollectionId).Distinct();
+            report.TotalItemCount = distinctItems.Select(x => collectionCountDictionary.TryGetValue(x, out var count) ? count : 0).Sum();
```

Comment:
I'm not sure the `TotalItemCount` still works - an item can be in multiple collections, so just summing the item count of each collection is going to contain duplicates.